packages/x-components/src/x-modules/facets-next/components/lists/__tests__/filters-injection.mixin.spec.ts

```diff
@@ -0,0 +1,189 @@
+import Vue, { ComponentOptions } from 'vue';
+import { Component, Prop } from 'vue-property-decorator';
+import {
+  createNextHierarchicalFilter,
+  createNextSimpleFilter
+} from '../../../../../__stubs__/filters-stubs.factory';
+import { XProvide } from '../../../../../components/decorators/injection.decorators';
+import FiltersInjectionMixin from '../filters-injection.mixin';
+
+@Component({
+  template: `
+    <div>
+    <slot/>
+    </div>`
+})
+class Provider extends Vue {
+  @Prop()
+  @XProvide('filters')
+  public injectedData!: any;
+}
+
+const Filters: ComponentOptions<Vue> = {
+  mixins: [FiltersInjectionMixin],
+  template: `
+    <div>
+    <li v-for="filter of renderedFilters" :key="filter.id">{{ filter.id }}</li>
+    </div>
+  `
+};
+
+describe('Test filters injection mixin', () => {
+  function renderFiltersMixin({
+    propFilters,
+    injectedFilters,
+    parentId
+  }: Partial<{
+    propFilters: Filter[];
+    injectedFilters: Filter[];
+    parentId: Filter['id'];
+  }> = {}): TestFilterInjectionAPI {
+    const wrapper = mount(
+      {
+        template: `
+          <Provider :injectedData="injectedFilters">
+          <Filters :filters="propFilters" :parentId="parentId"/>
+          </Provider>
+        `,
+        components: { Provider, Filters }
+      },
+      {
+        data() {
+          return {
+            propFilters,
+            injectedFilters,
+            parentId
+          };
+        }
+      }
+    );
+
+    return {
+      wrapper,
+      getRenderedFilterIds(): string[] {
+        return wrapper.findAll('li').wrappers.map(element => element.text());
+      },
+      async setPropFilters(propFilters: Filter[]) {
+        await wrapper.setData({ propFilters });
+      }
+    };
+  }
+
+  it('renders the props filters', async () => {
+    const whiteFilter = createNextSimpleFilter('color', 'white');
+    const blackFilter = createNextSimpleFilter('color', 'black');
+    const blueFilter = createNextSimpleFilter('color', 'blue');
+    const { setPropFilters, getRenderedFilterIds } = renderFiltersMixin({
+      propFilters: [whiteFilter, blackFilter]
+    });
+    let renderedIds = getRenderedFilterIds();
+
+    expect(renderedIds).toContain(whiteFilter.id);
+    expect(renderedIds).toContain(blackFilter.id);
+
+    await setPropFilters([whiteFilter, blackFilter, blueFilter]);
+    renderedIds = getRenderedFilterIds();
+
+    expect(renderedIds).toContain(whiteFilter.id);
+    expect(renderedIds).toContain(blackFilter.id);
+    expect(renderedIds).toContain(blueFilter.id);
+  });
+
+  it('renders the injected filters', () => {
+    const whiteFilter = createNextSimpleFilter('color', 'white');
+    const blackFilter = createNextSimpleFilter('color', 'black');
+    const blueFilter = createNextSimpleFilter('color', 'blue');
+    const { getRenderedFilterIds } = renderFiltersMixin({
+      injectedFilters: [whiteFilter, blackFilter, blueFilter]
+    });
+    let renderedIds = getRenderedFilterIds();
```

Comment:
This is never reassigned, change to `const`

```suggestion
    const renderedIds = getRenderedFilterIds();
```

- [ ] Change to `const`